Backpack-SwiftUI/AppSearchModal/Classes/BPKAppSearchModal.swift

```diff
@@ -74,7 +75,6 @@ public struct BPKAppSearchModal: View {
         .padding(.top, .base)
         .padding(.bottom, BPKSpacing.none)
         .background(.surfaceDefaultColor)
-        .accessibilityAddTraits(.isModal)
```

Comment:
The child views could not be accessed by identifier, blocking it for test automation.
|      Before      |     After    |
|---------------------|------------------|
|      <img width="300" alt="Screenshot 2024-05-06 at 14 41 34" src="https://github.com/Skyscanner/backpack-ios/assets/159815080/7fffb0f1-4059-4d80-892f-4db448f5659d">         |        <img width="300" alt="Screenshot 2024-05-06 at 15 33 08" src="https://github.com/Skyscanner/backpack-ios/assets/159815080/e51ba7a1-84f8-435e-b261-1b802b0d0db6">       |

Reply:
How does this work for screen readers after this change. Is the focus properly 'trapped' in the view once it's open? 

I assume it is because we present it with a .sheet{ } modifier, but could you double check please?

Reply:
There is no change for the current configuration of how our screens are placed in navigation structure. This modifier is responsible for preventing any undesirable screen readings of elements on the same level as the modal screen. It's still weird how Maestro reads the hierarchy here, and it seems like a small bug to me. However, it's completely fine to remove this modifier here, as it's not anything mandatory, it was just preventative.

Also, I additionally checked that with the accessibility inspector and there is no changes here.